sled-agent/src/services.rs

```diff
@@ -226,11 +215,42 @@ impl Config {
             sled_id,
             sidecar_revision,
             gateway_address,
-            all_svcs_config_path: default_services_config_path(),
+            all_svcs_ledger_path: default_services_ledger_path(),
+            storage_svcs_ledger_path: default_storage_services_ledger_path(),
         }
     }
 }
 
+// The filename of ServiceManager's internal storage.
+const SERVICES_CONFIG_FILENAME: &str = "services.toml";
+const STORAGE_SERVICES_CONFIG_FILENAME: &str = "storage-services.toml";
+
+// The default path to service configuration
+fn default_services_ledger_path() -> PathBuf {
+    Path::new(omicron_common::OMICRON_CONFIG_PATH)
+        .join(SERVICES_CONFIG_FILENAME)
+}
+
+// The default path to storage service configuration
+fn default_storage_services_ledger_path() -> PathBuf {
+    Path::new(omicron_common::OMICRON_CONFIG_PATH)
+        .join(STORAGE_SERVICES_CONFIG_FILENAME)
+}
+
+// TODO(ideas):
+// - "ServiceLedger"
+// - Manages the serializable "AllZoneRequests" object
+// - Constructor which reads from config location (kinda like
+// "read_from")
+// - ... Writer which *knows the type* to be serialized, so can direct it to the
+// appropriate output path.
+//
+// - TODO: later: Can also make the path writing safer, by...
```

Comment:
Maybe move this technique out into an issue?

Reply:
See: https://github.com/oxidecomputer/omicron/pull/2972 - I just finished implementing this.

Reply:
Oh. Excellent!